src/test/java/org/jenkinsci/plugins/workflow/steps/durable_task/ShellStepTest.java

```diff
@@ -692,6 +696,36 @@ private static final class HelloNote extends ConsoleNote<Run<?, ?>> {
         j.waitForMessage(new ExecutorStepExecution.RemovedNodeCause().getShortDescription(), b);
     }
 
+    @Test public void permanentlyDisconnectingAgentIsFatal() throws Exception {
+        DurableTaskStep.WATCHING_RECURRENCE_PERIOD = Duration.ofSeconds(30).toMillis();
+        logging.record(DurableTaskStep.class, Level.FINE);
+        var s = j.createSlave("remote", null, null);
+        s.setRetentionStrategy(RetentionStrategy.NOOP);
+        var p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("node('remote') {isUnix() ? sh('sleep 1000000') : bat('ping -t 127.0.0.1 > nul')}", true));
+        var b = p.scheduleBuild2(0).waitForStart();
+        j.waitForMessage(Functions.isWindows() ? ">ping" : "+ sleep", b);
+        s.toComputer().disconnect(null);
+        j.assertBuildStatus(Result.ABORTED, j.waitForCompletion(b));
+        j.waitForMessage(new ExecutorStepExecution.RemovedNodeTimeoutCause().getShortDescription(), b);
+    }
+
+    @Test public void temporarilyDisconnectingAgentIsNotFatal() throws Exception {
+        DurableTaskStep.WATCHING_RECURRENCE_PERIOD = Duration.ofSeconds(30).toMillis();
+        logging.record(DurableTaskStep.class, Level.FINE);
+        var s = j.createSlave("remote", null, null);
+        s.setRetentionStrategy(RetentionStrategy.NOOP);
+        var p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("node('remote') {isUnix() ? sh('sleep 60') : bat('ping -c 60 -t 127.0.0.1 > nul')}", true));
```

Comment:
DO you want to ping 60 times?  
`-c` is for routing and requires admin access.

```
❯ where ping
C:\Windows\System32\PING.EXE

  D:\s\github  ☸ docker-desktop default  where ping^C                                                   13:51:00 
❯ ping -c 60 -t 127.0.0.1
Access denied. Option -c requires administrative privileges.

❯ ping --help
Bad option --help.

Usage: ping [-t] [-a] [-n count] [-l size] [-f] [-i TTL] [-v TOS]
            [-r count] [-s count] [[-j host-list] | [-k host-list]]
            [-w timeout] [-R] [-S srcaddr] [-c compartment] [-p]
            [-4] [-6] target_name

Options:
    -t             Ping the specified host until stopped.
                   To see statistics and continue - type Control-Break;
                   To stop - type Control-C.
    -a             Resolve addresses to hostnames.
    -n count       Number of echo requests to send.
    -l size        Send buffer size.
    -f             Set Don't Fragment flag in packet (IPv4-only).
    -i TTL         Time To Live.
    -v TOS         Type Of Service (IPv4-only. This setting has been deprecated
                   and has no effect on the type of service field in the IP
                   Header).
    -r count       Record route for count hops (IPv4-only).
    -s count       Timestamp for count hops (IPv4-only).
    -j host-list   Loose source route along host-list (IPv4-only).
    -k host-list   Strict source route along host-list (IPv4-only).
    -w timeout     Timeout in milliseconds to wait for each reply.
    -R             Use routing header to test reverse route also (IPv6-only).
                   Per RFC 5095 the use of this routing header has been
                   deprecated. Some systems may drop echo requests if
                   this header is used.
    -S srcaddr     Source address to use.
    -c compartment Routing compartment identifier.
    -p             Ping a Hyper-V Network Virtualization provider address.
    -4             Force using IPv4.
    -6             Force using IPv6.


```


```suggestion
        p.setDefinition(new CpsFlowDefinition("node('remote') {isUnix() ? sh('sleep 60') : bat('ping -r 60 -t 127.0.0.1 > nul')}", true));
```

Reply:
TIL `timeout.exe`  https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/timeout  should be available in Windows Server also.

(untested other than the command runs locally)

```suggestion
        p.setDefinition(new CpsFlowDefinition("node('remote') {isUnix() ? sh('sleep 60') : bat('timeout /T 60 /NOBREAK')}", true));
```